pkg/descheduler/evictions/evictions.go

```diff
@@ -137,7 +140,48 @@ func (pe *PodEvictor) EvictPod(ctx context.Context, pod *v1.Pod, node *v1.Node,
 		return false, fmt.Errorf("Maximum number %v of evicted pods per %q namespace reached", *pe.maxPodsToEvictPerNamespace, pod.Namespace)
 	}
 
-	err := evictPod(ctx, pe.client, pod, pe.policyGroupVersion)
+	if pe.ignoreLocalPvcPods {
+		klog.V(1).InfoS("checking if pod has Local PVC Storage")
+		isPodWithLocalPVC, _ := utils.IsPodWithLocalPVC(ctx, pe.client, pod)
+		if isPodWithLocalPVC {
+			return false, fmt.Errorf("This pod %q/%q has Local PVC Storage and ignoreLocalPvcPods is set to true", pod.Namespace, pod.Name)
+		}
+	}
+
+	// only check for and delete PVC if
+	// We want to evict PVC pods and
+	// We want to evict pod and delete local PVC
+	if !pe.ignoreLocalPvcPods && !pe.ignorePvcPods {
+		klog.V(1).InfoS("checking if pod has Local PVC Storage")
+		isPodWithLocalPVC, err := utils.IsPodWithLocalPVC(ctx, pe.client, pod)
+		if err != nil {
+			klog.InfoS("Error checking for Local PVC for pod", "pod", klog.KObj(pod), "error", err)
+			return false, err
+		}
+		if isPodWithLocalPVC {
+			klog.V(1).InfoS("Pod HAS Local PVC Storage")
+		} else {
+			klog.V(1).InfoS("Pod does NOT have Local PVC Storage")
+		}
+		if isPodWithLocalPVC {
+			// dry run first to see if we should delete the PVC
+			klog.V(1).InfoS("Pod has Local PVC, performing dryrun eviction")
+			dryRunErr := evictPod(ctx, pe.client, pod, pe.policyGroupVersion, true)
```

Comment:
don't need a new variable, just use ` evictPod(ctx, pe.client, pod, pe.policyGroupVersion, true)` is enough

Reply:
Thanks, I simplified it a bit